tikv/raftstore/snap_builder.go

```diff
@@ -0,0 +1,231 @@
+package raftstore
+
+import (
+	"bytes"
+	"os"
+
+	"github.com/coocood/badger"
+	"github.com/ngaut/unistore/lockstore"
+	"github.com/ngaut/unistore/rocksdb"
+	"github.com/ngaut/unistore/tikv/mvcc"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/kvrpcpb"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/tidb/util/codec"
+	"math"
+)
+
+func newSnapBuilder(cfFiles []*CFFile, db *badger.DB, lockStore *lockstore.MemStore, region *metapb.Region) (*snapBuilder, error) {
+	b := new(snapBuilder)
+	b.cfFiles = cfFiles
+	b.endKey = rawRegionKey(region.EndKey)
+	b.txn = db.NewTransaction(false)
+	b.dbIterator = b.txn.NewIterator(badger.DefaultIteratorOptions)
+	startKey := rawRegionKey(region.StartKey)
+	b.dbIterator.Seek(startKey)
+	if b.dbIterator.Valid() && !b.reachEnd(b.dbIterator.Item().Key()) {
+		b.curDBKey = b.dbIterator.Item().Key()
+	}
+	b.dbOldIterator = b.txn.NewIterator(badger.DefaultIteratorOptions)
+	b.dbOldIterator.Seek(mvcc.EncodeOldKey(startKey, math.MaxUint64))
```

Comment:
I don't understand how to distinguish old iterator and latest iterator since they are built with the same API NewIterator.

Reply:
They seek different ranges.

Reply:
Got it, so the old values has a different prefix?